### ISSUE TITLE
Inconsistent use of clip paths between IM 6 and IM 7

### ISSUE THREAD

**Issue by fmw42** [ImageMagick/ImageMagick#2576]
I am getting inverted transparency from clip paths on tiff file between IM 6.9.11-29 and IM 7.0.10.29 Q16  Mac OSX Sierra

Input:

https://github.com/ImageMagick/ImageMagick/files/5222811/bucket.zip

IM 6

`convert -quiet bucket.tif[0] -alpha transparent -clip -alpha opaque -strip result6.png`

![result6](https://user-images.githubusercontent.com/13458218/93175398-6a9ef480-f6e4-11ea-9fc1-0ed25629ff16.png)

IM 7

`magick -quiet bucket.tif[0] -alpha transparent -clip -alpha opaque -strip result7a.png`

![result7a](https://user-images.githubusercontent.com/13458218/93175418-71c60280-f6e4-11ea-8e3a-d1d66698e82f.png)

IM 7 workaround (negate alpha channel)

`magick -quiet bucket.tif[0] -alpha transparent -clip -alpha opaque -strip -channel a -negate +channel result7b.png`

![result7b](https://user-images.githubusercontent.com/13458218/93175428-74c0f300-f6e4-11ea-8ba4-a7f287520e4e.png)

**Comment by urban-warrior** [comment]
Curious.  We get expected results without the need to negate with ImageMagick 7.0.10-30 Beta.

**Comment by fmw42** [comment]
You probably fixed it in -30 beta :)

**Comment by fmw42** [comment]
I just downloaded the latest beta and it still needs inverting of the alpha channel using IM 7.0.10.30 beta on my Mac OSX Sierra.

**Comment by FredrikEngquist** [comment]
I have also seen this.
With IM7 I have had to change '-clip-path "XYZ"' to '+clip-path "XYZ"' which for some curious reason de-negates the path.
(instead of dropping it as + usually does.)

**Comment by fmw42** [comment]
This is still an issue for me with IM 7.0.10.30 on Mac OSX Sierra where -clip produces an inverted transparency.  I also note that using +clip rather than -clip produces the expected result on IM 7 similar to IM 6. On IM 6, both -clip and +clip produce the same expected result.

**Comment by fmw42** [comment]
This is still an issue on IM 7.1.0.6

**Comment by urban-warrior** [comment]
Note the alpha channel has reverse polarity from ImageMagick version 6.  See https://imagemagick.org/script/porting.php.  To get the same clipping behavior from an alpha channel in IM6 and IM7, it is expected that you must negate the alpha channel.  Its possible there was a bug in an earlier release of IM7 that generated the same clipping results-- and that bug is now fixed to get expected behavior.

**Comment by fmw42** [comment]
I do not understand this. It seems that the reversal and need to negate is only apparent in clip paths. It is not reversed when doing -compose copy_opacity nor are the transparency values different between IM 6 and IM 7. So why only with the clip path?

**Comment by urban-warrior** [comment]
Try convert `bucket.tif` to `bucket.pam`.  Do you then get expected results from the clipping using `bucket.pam`?  `bucket.tif` includes a clipping path in the embedded Photoshop profile that is likely influencing the clipping behavior.  Clipping, by the way, is just a write mask which differs from how clipping is handled by IMv6.

**Comment by fmw42** [comment]
```
magick bucket.tif bucket.pam

magick -quiet bucket.pam[0] -alpha transparent -clip -alpha opaque -strip result7a.png

```

**magick: no clip path defined 'bucket.pam': Undefined error: 0 @ error/image.c/ClipImagePath/721.**

**Comment by urban-warrior** [comment]
Got it.  The alpha polarity between IM6 and IM7 is reversed.  It appears its the same for the clipping path-- which generate a write mask.  We'll add a patch to fix this problem and generate a new beta.

**Comment by fmw42** [comment]
OK. This seems to be fixed in IM 7.1.0-7
